x-pack/test/functional/apps/code/history.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { TestInvoker } from './lib/types';
+
+// eslint-disable-next-line import/no-default-export
+export default function manageRepositoriesFunctionalTests({
+  getService,
+  getPageObjects,
+}: TestInvoker) {
+  // const esArchiver = getService('esArchiver');
+  const testSubjects = getService('testSubjects');
+  const retry = getService('retry');
+  const log = getService('log');
+  const browser = getService('browser');
+  const queryBar = getService('queryBar');
+  const config = getService('config');
+  const FIND_TIME = config.get('timeouts.find');
+  const find = getService('find');
+  const PageObjects = getPageObjects(['common', 'header', 'security', 'code', 'home']);
+
+  describe('Code', () => {
+    const repositoryListSelector = 'codeRepositoryList codeRepositoryItem';
+
+    describe('browser history can go back while exploring code app', () => {
+      before(async () => {
+        // Navigate to the code app.
+        await PageObjects.common.navigateToApp('code');
+        await PageObjects.header.waitUntilLoadingHasFinished();
+      });
+      // after(async () => await esArchiver.unload('code'));
+
+      after(async () => {
+        await PageObjects.security.logout();
+      });
+
+      it('import repository', async () => {
```

Comment:
I think we have already discussed this. There are 3 pages now, please cover

1. from AdminPage to SourceViewPage, cover the cases of going back and forward
2. from SourceViewPage to SearchPage, cover the cases of going back and forward
3. from SearchPage to SourceViewPage, cover the cases of going back and forward
4. from one SourceViewPage to another SourceViewPage by jumping to the definition, cover the cases of going back and forward.

Each test should be self-contained, instead of relying on the operations from a previous test, like you do in here. This's not quite stable.